packages/js/src/executors/verdaccio/verdaccio.impl.ts

```diff
@@ -172,37 +173,72 @@ function setupNpm(options: VerdaccioExecutorSchema) {
 
 function setupYarn(options: VerdaccioExecutorSchema) {
   try {
-    execSync('yarn --version');
-  } catch (e) {
-    return () => {};
-  }
+    const yarnV1 = major(execSync('yarn --version').toString().trim()) === 1;
 
-  let yarnRegistryPath;
-  try {
-    yarnRegistryPath = execSync(`yarn config get registry`)
-      ?.toString()
-      ?.trim()
-      ?.replace('\u001b[2K\u001b[1G', ''); // strip out ansi codes
-    execSync(`yarn config set registry http://localhost:${options.port}/`);
-    logger.info(`Set yarn registry to http://localhost:${options.port}/`);
-  } catch (e) {
-    throw new Error(
-      `Failed to set yarn registry to http://localhost:${options.port}/: ${e.message}`
-    );
-  }
-
-  return () => {
     try {
-      if (yarnRegistryPath) {
-        execSync(`yarn config set registry ${yarnRegistryPath}`);
-        logger.info(`Reset yarn registry to ${yarnRegistryPath}`);
-      } else {
-        execSync(`yarn config delete registry`);
+      const registryConfigName = yarnV1 ? 'registry' : 'npmRegistryServer';
+
+      const yarnRegistryPath = execSync(`yarn config get ${registryConfigName}`)
+        ?.toString()
+        ?.trim()
+        ?.replace('\u001b[2K\u001b[1G', ''); // strip out ansi codes
+
+      execSync(
+        `yarn config set ${registryConfigName} http://localhost:${options.port}/` +
+          options.location ===
+          'home'
+          ? ' --home'
+          : ''
+      );
+
+      logger.info(`Set yarn registry to http://localhost:${options.port}/`);
+
+      if (!yarnV1) {
+        execSync(
+          `yarn config set unsafeHttpWhitelist --json '["localhost"]'` +
```

Comment:
This will overwrite the existing value and we don't have a stored value to set it back to, potentially breaking folks.